sdk/storage/azure-storage-file/src/main/java/com/azure/storage/file/FileClient.java

```diff
@@ -322,11 +463,33 @@ public VoidResponse delete() {
      *
      * @param data The data which will upload to the storage file.
      * @param length Specifies the number of bytes being transmitted in the request body. When the FileRangeWriteType is set to clear, the value of this header must be set to zero..
-     * @return A response that only contains headers and response status code
+     * @return The {@link FileUploadInfo file upload info}
+     * @throws StorageErrorException If you attempt to upload a range that is larger than 4 MB, the service returns status code 413 (Request Entity Too Large)
+     */
+    public FileUploadInfo upload(ByteBuf data, long length) {
+        return uploadWithResponse(data, length, Context.NONE).value();
+    }
+
+    /**
+     * Uploads a range of bytes to the beginning of a file in storage file service. Upload operations performs an in-place write on the specified file.
+     *
+     * <p><strong>Code Samples</strong></p>
+     *
+     * <p>Upload "default" to the file. </p>
+     *
+     * {@codesnippet com.azure.storage.file.fileClient.uploadWithResponse#flux-long}
+     *
+     * <p>For more information, see the
+     * <a href="https://docs.microsoft.com/en-us/rest/api/storageservices/put-range">Azure Docs</a>.</p>
+     *
+     * @param data The data which will upload to the storage file.
+     * @param length Specifies the number of bytes being transmitted in the request body. When the FileRangeWriteType is set to clear, the value of this header must be set to zero..
+     * @param context Additional context that is passed through the Http pipeline during the service call.
+     * @return A response containing the {@link FileUploadInfo file upload info} with headers and response status code
      * @throws StorageErrorException If you attempt to upload a range that is larger than 4 MB, the service returns status code 413 (Request Entity Too Large)
      */
-    public Response<FileUploadInfo> upload(ByteBuf data, long length) {
-        return fileAsyncClient.upload(Flux.just(data), length).block();
+    public Response<FileUploadInfo> uploadWithResponse(ByteBuf data, long length, Context context) {
```

Comment:
This should be upload and remove the `Context`

Reply:
I think we need to have this overload as well since `range` calculation is different when offset is specified.
This is still an existing call on the Async API client.